src/anndata/_core/sparse_dataset.py

```diff
@@ -358,10 +361,12 @@ class BaseCompressedSparseDataset(ABC):
     format: Literal["csr", "csc"]
     """The format of the sparse matrix."""
     _group: GroupStorageType
+    _shall_use_sparse_array_on_read: bool
 
     def __init__(self, group: GroupStorageType):
         type(self)._check_group_format(group)
         self._group = group
+        self._shall_use_sparse_array_on_read = settings.shall_use_sparse_array_on_read
```

Comment:
Up for debate whether this should be dynamic or not.  I think once the class is initiated, so too should this property

Reply:
I think it would be pretty surprising if the setting “does nothing”

And if someone changes this setting in the middle of a loop while they’re operating on an AnnData object, I think they’re kinda asking for weird things to happen?

Reply:
Sorry, not sure I follow.  Those seem like like they are opposites.  If someone changes the setting while holding on to the object still, you think the behavior of the object (`CS{R,C}Dataset`) should change?  Or the opposite, that it should not because that is asking for trouble?

Reply:
I mean code like this: what would you expect `m` to be?

```py
adata = ad.read_backed(...)  # this line would lock in `adata.X._shall_use_sparse_array_on_read`

# several notebook cells later
subset = adata[adata.obs["sample"] == 1]
with ad.settings.override(shall_use_sparse_array_on_read=True):
    m = subset.X.to_memory()
```

Reply:
Ok, this makes sense.  Should be dynamic then!